app/api/groups/get_group_progress.go

```diff
@@ -0,0 +1,175 @@
+			)`).
+		Where("main_attempts.idItem IN (?)", itemIDs). // preliminary filter (makes the query faster)
+		Group("main_attempts.idGroup, main_attempts.idItem")
+
+	endMembers := srv.Store.GroupAncestors().
+		Select("groups_ancestors.idGroupAncestor AS idGroup, end_member.ID AS idEndMember").
+		Joins(`
+			JOIN groups AS end_member
+			ON end_member.ID = groups_ancestors.idGroupChild AND end_member.sType IN ('UserSelf', 'Team')`).
+		// The team users filtering starts here (is not well specified yet)
+		Joins(`
+			LEFT JOIN groups_groups AS team_link
+			ON
+				end_member.ID IS NOT NULL AND
+				end_member.sType != 'Team' AND
+				team_link.idGroupChild = end_member.ID AND
+				team_link.sType IN('invitationAccepted','requestAccepted','direct')`).
+		Joins(`
+			LEFT JOIN groups_ancestors AS team_ancestor_check
+			ON team_ancestor_check.idGroupChild = team_link.idGroupParent AND
+				team_ancestor_check.idGroupAncestor = groups_ancestors.idGroupAncestor`).
+		Joins("LEFT JOIN groups AS team ON team.ID = team_ancestor_check.idGroupChild AND team.sType = 'Team'").
+		Joins(`
+			LEFT JOIN (SELECT 1 as bHasOtherParents) AS other_parents
+			ON team.ID IS NOT NULL AND (
+				SELECT 1
+				FROM groups_groups
+				WHERE idGroupChild = end_member.ID AND
+					idGroupParent != team.ID AND
+					idGroupParent IN (SELECT idGroupChild FROM groups_ancestors AS ga WHERE ga.idGroupAncestor = groups_ancestors.idGroupAncestor)
+				LIMIT 1
+			) = 1`).
+		// The team users filtering ends here
+		Where("groups_ancestors.idGroupAncestor IN (?)", ancestorGroupIDs). // preliminary filter (makes the query faster)
+		Where("groups_ancestors.idGroupAncestor != groups_ancestors.idGroupChild").
+		Group("groups_ancestors.idGroupAncestor, end_member.ID").
+		// The team users filtering (is not well specified yet)
+		Having("MIN(end_member.sType = 'Team' OR (team.ID IS NULL OR (team.ID IS NOT NULL AND other_parents.bHasOtherParents)))")
+
+	// We want to keep groups that don't have end members
+	groupsEndMembers := srv.Store.Groups().
+		Select("groups.ID AS idGroup, end_member.idEndMember AS idEndMember").
+		Joins(`
+			LEFT JOIN ? AS end_member
+			ON end_member.idGroup = groups.ID`, endMembers.SubQuery()).
+		Where("groups.ID IN (?)", ancestorGroupIDs)
+
+	var dbResult []map[string]interface{}
+	service.MustNotBeError(srv.Store.Raw(`
+		SELECT
+			groups_end_members.idGroup AS idGroup,
+			items.ID AS idItem,
+			AVG(IFNULL(end_members_stats.iMaxEndMemberScore, 0)) AS iAverageScore,
+			AVG(IFNULL(end_members_stats.bMaxEndMemberValidated, 0)) AS iValidationRate,
+			AVG(IFNULL(end_members_stats.nbEndMemberBestScoreHintsCached, 0)) AS iAvgHintsRequested,
+			AVG(IFNULL(end_members_stats.nbEndMemberBestScoreSubmissionAttempts, 0)) AS iAvgSubmissionsAttempts,
+			AVG(IFNULL(end_members_stats.sEndMemberTimeSpent, 0)) AS sAvgTimeSpent
+		FROM ? AS groups_end_members
+		JOIN items FORCE INDEX(PRIMARY) ON items.ID IN (?)
```

Comment:
Do you need to join the items? It seems you are just using the IDs.
BTW, haven't you already filtered the items in `endMembersStats` ?

Reply:
`endMembersStats` has gone. It is possible to get rid of `items` here. It will look like ```JOIN (SELECT 123 AS ID UNION ALL SELECT 124 UNION ALL SELECT 125 ...) AS items```

Reply:
fixed